components/Common/AvatarGroup/Avatar/index.tsx

```diff
@@ -0,0 +1,29 @@
+import type { FC } from 'react';
+import * as RadixAvatar from '@radix-ui/react-avatar';
+import styles from './index.module.css';
+
+type AvatarProps = {
+  src?: string;
+  alt?: string;
+  extraClass?: string;
+  onClick?: () => void;
+};
+
+const usrnameToFallbackText = (name?: string) => {
+  // sanitize name before using it
+  if (!name) return;
+  const sanitizeName = name.toLowerCase().replace(' ', '-');
```

Comment:
This whole code can be extremely simplified to:

```ts
const fallbackAltAsAvatar = (alt: string = '') => 
  alt.match(/\b(\w)/g).join('').substring(0, 2)
```

The new name is also more inline with what this code does... As it's not grabbing an username, but the `alt` text.

Reply:
it's will become to much complicated to read because prettier taking one line for each "dot". and regex possibly null.

Reply:
Wdym? The code I placed here does the exact same as your code above. It's just a simplification, please apply it, and/or add unit tests showing what is your expected behaviour and what is not working correctly.

You can always tweak the regex. And also you could "document" the regex if you want.

Reply:
OK i will deep dive (regex is my weak point)

Reply:
ChatGPT is your friend. "Make me a regex in JavaScript that does XXXX by using the String.match function"